File: features/offices/src/main/java/com/nivasafinance/features/offices/exception/OfficeNotFoundException.java
package com.nivasafinance.features.offices.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import com.nivasafinance.common.exception.ResourceNotFoundException;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class OfficeNotFoundException extends ResourceNotFoundException {

    @Serial
    private static final long serialVersionUID = 1L;

    public OfficeNotFoundException(String key, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(
                "error.office.key.not.found",
                new Object[]{key},
                messageSource
        ));
    }
}

File: features/offices/src/main/java/com/nivasafinance/features/offices/exception/OfficeNotFoundException.kt
package com.nivasafinance.features.offices.exception

import exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import java.util.Locale
import java.util.UUID

class OfficeNotFoundException(message: String) : ResourceNotFoundException(message) {
    companion object {
        private val locale: Locale = LocaleContextHolder.getLocale()
        const val KEY = "error.office.id.not.found"
    }

    constructor(id: UUID, messageSource: MessageSource) : this(
        messageSource.getMessage(KEY, arrayOf(id), locale)
    )
}
